Count up, set background red, and ring when timer completes.

emosenkis opened the pull request:
![InfiniSim_2022-12-28_222533](https://user-images.githubusercontent.com/1794126/209880757-704fffa1-cd31-4164-bb58-8cffe27f5b1b.gif)

Fixes #1374
Fresh implementation of similar features of #557

Riksu9000 reviewed (commented):
There is an issue that when closing the app, the alerting continues. I think pressing the button once should stop the alert, but stay in the app, while long pressing the button should go to the watchface and also stop the alert.

Alarm (app) keeps the watch awake during the alerting. I wonder if this should do the same.

If left unattended, the alert can go on forever, making noise and draining the battery. The vibration should stop after ~10s and the watch can then go to sleep, but keep counting the overtime.

emosenkis commented on src/displayapp/widgets/Counter.h:
```diff
@@ -27,6 +27,10 @@ namespace Pinetime {
+        lv_style_t* GetLabelStyle() {
+          return &labelStyle;
+        }
+
```
Done

emosenkis reviewed (commented):
> There is an issue that when closing the app, the alerting continues. I think pressing the button once should stop the alert, but stay in the app, while long pressing the button should go to the watchface and also stop the alert.
> 
> Alarm (app) keeps the watch awake during the alerting. I wonder if this should do the same.
> 
> If left unattended, the alert can go on forever, making noise and draining the battery. The vibration should stop after ~10s and the watch can then go to sleep, but keep counting the overtime.

For now, I've made it keep the watch awake and vibrate for 10s, after which it stops. Can your suggestions about the two button behaviors be separate from this PR? I think this is already a big improvement and should enable many more real-world use cases for the timer now that it will be easier to notice when it completes, and it counts up after completing.

On a separate note, I had to add another timer object to TimerController - I tried to set the existing timer for 10s and restart it in `OnTimerEnd` but while both calls succeeded, the callback didn't get called again.

Riksu9000 commented:
Adding the button behaviour shouldn't be too complicated and it's pretty important to avoid making the stopping of the ringing hinder usability. Also I consider the fact that the vibration doesn't stop when closing the app a bug.

I think the timed stopping of the ringing should probably be handled in the app rather than the controller, as I think it's a UI thing. You can use LVGL tasks for this. Please refer to AlarmController.cpp and Alarm.cpp for both of these issues.

emosenkis commented:
@Riksu9000 I think I've addressed all your feedback - the button now behaves the way you asked, the app controls the ringing, leaving the app by button or by swiping should ensure that the ringing is stopped.

emosenkis commented:
@Riksu9000 rebased and fixed

emosenkis commented:
Ping @Riksu9000

Riksu9000 commented:
Long pressing hasn't been addressed and there are some issues still, but I'm also reconsidering this feature in this issue https://github.com/InfiniTimeOrg/InfiniTime/issues/391#issuecomment-1383231455
